src/jenkins/BuildManifest.groovy

```diff
@@ -25,12 +25,56 @@ class BuildManifest implements Serializable {
         String getFilename() {
             return this.name.toLowerCase().replaceAll(' ', '-')
         }
+
+        String getPackageName() {
+            String packagePrefix = [
+                this.getFilename(),
+                this.version,
+                this.platform,
+                this.architecture,
+            ].join('-')
+            return packagePrefix + '.tar.gz'
+        }
+    }
+
+
+
+    class Components extends HashMap<String , Component> {
+
+        Components(ArrayList data) {
+            data.each {
+                item ->
+                    Component component = new Component(item)
+                    (this[component.name] = component)
```

Comment:
The parenthesis are unnecessary. You're assigning the component at the key `component.name`. You could also add a method called `add(Component component)` and the code becomes cleaerer:

```groovy
Components(ArrayList data) {
   data.each { item-> this.add(new Component(item)) }
}

public void add(Component component) {
    this[component.name] = component
}
```

Reply:
Removed the parenthesis for now